### PR TITLE
pybind/mgr/*: fix config_notify handling of default values

### PR DESCRIPTION
**Pull request by smithfarm**
After 5ace82e65c72847fb875fc01c419937a26a59d70 was merged, I found
three more instances of the code being patched.

The commit message of 5ace82e65c72847fb875fc01c419937a26a59d70 was/is:

"The default values are handled by mgr_module.py's _get_module_option();
the or here means that we break any non-true (0, false, none) value and
override it with the default."

Fixes: 5ace82e65c72847fb875fc01c419937a26a59d70
Fixes: https://tracker.ceph.com/issues/43746
Signed-off-by: Nathan Cutler <ncutler@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/diskprediction_cloud/module.py [hunk 1]
```diff
@@ -246,7 +246,7 @@ def refresh_config(self):
         for opt in self.MODULE_OPTIONS:
             setattr(self,
                     opt['name'],
-                    self.get_module_option(opt['name']) or opt['default'])
+                    self.get_module_option(opt['name']))
```
Comment (smithfarm) [inline src/pybind/mgr/diskprediction_cloud/module.py]: note: this one is in `refresh_config`, not `config_notify`

File: src/pybind/mgr/diskprediction_local/module.py [hunk 2]
```diff
@@ -63,7 +63,7 @@ def refresh_config(self):
         for opt in self.MODULE_OPTIONS:
             setattr(self,
                     opt['name'],
-                    self.get_module_option(opt['name']) or opt['default'])
+                    self.get_module_option(opt['name']))
```
Comment (smithfarm) [inline src/pybind/mgr/diskprediction_local/module.py]: also in `refresh_config`

**Comment by smithfarm** [comment]
retargeted to octopus

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2020-03-25_08:14:01-rados-wip-kefu-testing-2020-03-25-1220-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2020-03-25_10:52:52-fs-wip-kefu-testing-2020-03-25-1220-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2020-03-25_10:54:21-rgw-wip-kefu-testing-2020-03-25-1220-distro-basic-smithi/

**Pull request merged**